### PR TITLE
Fix nn.Module.apply doc formatting

### PR DESCRIPTION
**Pull request by ssnl**

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ def apply(self, fn):
         Returns:
             Module: self
 
-        Example:
+        Example::
```
Comment (fmassa) [inline torch/nn/modules/module.py]: When should we have `Example:` or `Example::`? I see that you have used `Example:` in other parts of this PR

Comment (SsnL) [reply torch/nn/modules/module.py]: I think such blocks are called [literal blocks (fully minimized form)](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#literal-blocks). So they should all use double-colon really. But the main problem with this is that it is missing a newline.

File: torch/nn/modules/module.py [hunk 2]
```diff
@@ -619,6 +621,7 @@ def modules(self):
             Duplicate modules are returned only once. In the following
             example, ``l`` will be returned only once.
 
+        Example:
```
Comment (fmassa) [inline torch/nn/modules/module.py]: So in this case I suppose we should have a new line and two colons?

File: torch/nn/modules/module.py [hunk 3]
```diff
@@ -643,6 +646,7 @@ def named_modules(self, memo=None, prefix=''):
...
```
Comment (fmassa) [inline torch/nn/modules/module.py]: Same here?

**Comment by SsnL** [comment]
@fmassa Thanks! Fixed those as well :)

**Pull request merged**